main.py

```diff
@@ -0,0 +1,81 @@
+import doctest
+from graph import *
+
+INPUT_FILE = open("career.in", "r")
+OUTPUT_FILE = open("career.out", "w")
+
+
+def get_maximum_profit(hierarchy):
+    """
+    The function to implement a search for maximum value path
+    :param hierarchy: graph of pyramidal type
+    :return: maximum path weight
+    >>> get_maximum_profit(get_items())
+    13
+    >>> G = Graph()
+    >>> G.add_vertex(4, [])
+    >>> G.add_vertex(1, [])
+    >>> G.add_vertex(2, [0, 1])
+    >>> get_maximum_profit(G)
+    6
+    """
+    if len(hierarchy.vertexes):
+        hierarchy.breadth_search([len(hierarchy.vertexes) - 1], is_first=True)
```

Comment:
So you call this method only here. And obviously you want to pass just one value – why the hell is your signature only accepts array then?

Bonus question: why not just -1 instead of `len(hierarchy.vertexes) - 1?`

Reply:
Also, why not to return the value you're interested in? Why instead to store a state inside a class?

Reply:
Storing value inside of class is of course a nonsense for a method.
But it was more comfortable to do ion the stage of implementation, because of the ease of reaching this value.
However regarding OOP principles, it would be better to return a value from the method (maximum path length does not describe a part of graph and is also not a constant value)

Reply:
Yes and it's not only OOP. In C++ you can mark arguments as IN or OUT, in other languages you don't. A method is usually expected to change a state of self or return the value. In rare cases it changes a state of third party objects. Nevertheless if you think of usability of your code — returning is a good habit